### DISCUSSION TITLE
Tag creation trigger not working with lerna publish

### DISCUSSION THREAD

**Discussion by callafeucht** [community/community#25085] [Actions]
<p>We’re using <a href="https://github.com/lerna/lerna" rel="nofollow noopener">lerna</a> to manage versioning and releases of multiple packages in a monorepo. When we run the lerna publish command, new tags are created and pushed to our repo, with the commit attributed to the user who ran the lerna publish command on their machine. No matter what configuration we try in our workflow YAML definition file, the lerna publish command does not trigger the workflow. Manual creation and pushing of tags triggers the workflow as expected.</p>
<p>We’ve tried the following configurations:</p>
<pre><code>on:
  create:
    branches:
      - 'refs/tags/@orgname/**'

on:
  create:
    tags:
      - '**'

on:
  push:
    branches:
      - 'refs/tags/@orgname/**'

on:
  push:
    tags:
      - '**'
</code></pre>
<p>Has anyone else been able to trigger a GitHub action from the tags created and pushed by lerna publish? Am I missing something?</p>

**Comment by BrightRan** [comment, marked as answer]
<p>After executing the command <strong>lerna publish</strong> to create and push new tags to your GitHub repository, did you check if the tags have been pushed successfully to the repository on web UI? And how many tags did you create and push at once?</p>
<p>I checked the documentations about <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdeveloper.github.com%2Fv3%2Factivity%2Fevents%2Ftypes%2F%23createevent&amp;data=02%7C01%7Cv-brira%40microsoft.com%7C31269325cd2b4a236c1408d7934ef323%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637139839604564645&amp;sdata=bCSVSfYyY2dM7%2B3NsVypBw5VOw8dQXdkoP9I3JA9yEk%3D&amp;reserved=0" rel="nofollow noopener">create event</a> and <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdeveloper.github.com%2Fv3%2Factivity%2Fevents%2Ftypes%2F%23pushevent&amp;data=02%7C01%7Cv-brira%40microsoft.com%7C31269325cd2b4a236c1408d7934ef323%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637139839604564645&amp;sdata=wj7vYDfA7pJ1XzUX4QKBCK0JwY9PGBXGxCf0EfQV4Xc%3D&amp;reserved=0" rel="nofollow noopener">push event</a>, and found the below note:</p>
<p><strong>Note:</strong> You will not receive a webhook for this event when you push more than three tags at once.</p>
<p>Maybe this is why the workflows were not triggered.</p>
<p>I also tested with the git commands: use <strong>git tag &lt;tag_name&gt;</strong> to create new tags, and use <strong>git push --tags</strong> to push tags to GitHub repository. When the number of the tags I pushed at once was less than or equal to 3, it triggered a workflow run for each tag. When the number was more than 3, no workflow runs triggered.</p>

**Comment by callafeucht** [comment]
<p>The tags did show as pushed successfully to the repository; sorry for leaving that out of my initial post.</p>
<p>Thank you for that find – that is absolutely the issue here. Our lerna publish command is pushing a tag for every single package in the monorepo simultaneously (at least in these test cases), which is 23 tags at once. I tested locally with 3 manually created tags and triggered the workflow successfully.</p>
